README.md

```diff
@@ -1,67 +1,139 @@
 # cli-transparent-tunnel 
 
-* CLI wrapper that:
-  * create a ssl tunnel -> run the command through the tunnel -> close the tunnel.
-  * basically it can act as an inline replacer fot the CLI 
-    * the wrapper check if there is a tunnel configured for the required endpoint, if it has:
-      * it adds the ssh tunnel flags to the CLI and invoke the command with all args.
-      * if not, it just invokes the CLI 
+## Introduction
 
-* THIS PROJECT is WIP, I believe it can solve accessing k8s clusters through ssh tunnel in certain scenarios such as CI / CD etc .
-* Let me know what you think.
+SSH tunneling is a great method of transporting arbitrary networking data over an encrypted SSH connection.
+It can be used to add 
+- encryption to legacy applications. 
+- access network servers behind an ssh bastion server 
+- and more .. 
 
-## supported CLIs:
-  - kubectl
+To do so, we need to create an SSH tunnel and point the local client to the tunnel's local port. 
 
+for examples, running psql command through a ssl tunnel
```

Comment:
For example, consider the case you would like to access a PostgreSQL database located on a remote machine which blocks any port except from SSH (22) from the obvious security considerations.